script/hassfest/requirements.py

```diff
@@ -41,22 +41,116 @@
 PIP_REGEX = re.compile(r"^(--.+\s)?([-_\.\w\d]+.*(?:==|>=|<=|~=|!=|<|>|===)?.*$)")
 PIP_VERSION_RANGE_SEPARATOR = re.compile(r"^(==|>=|<=|~=|!=|<|>|===)?(.*)$")
 
-FORBIDDEN_PACKAGES = {"setuptools", "wheel"}
-FORBIDDEN_PACKAGE_EXCEPTIONS = {
-    # Direct dependencies
-    "fitbit",  # setuptools (fitbit)
-    "influxdb-client",  # setuptools (influxdb)
-    "microbeespy",  # setuptools (microbees)
-    "pyefergy",  # types-pytz (efergy)
-    "python-mystrom",  # setuptools (mystrom)
-    # Transitive dependencies
-    "arrow",  # types-python-dateutil (opower)
-    "asyncio-dgram",  # setuptools (guardian / keba / minecraft_server)
-    "colorzero",  # setuptools (remote_rpi_gpio / zha)
-    "incremental",  # setuptools (azure_devops / lyric / ovo_energy / system_bridge)
-    "pbr",  # setuptools (cmus / concord232 / mochad / nx584 / opnsense)
-    "pycountry-convert",  # wheel (ecovacs)
-    "unasync",  # setuptools (hive / osoenergy)
+FORBIDDEN_PACKAGES = {"codecov", "pytest", "setuptools", "wheel"}
+FORBIDDEN_PACKAGE_EXCEPTIONS: dict[str, dict[str, set[str]]] = {
+    # In the form dict("domain": {"package": {"reason1", "reason2"}})
+    # - domain is the integration domain
+    # - package is the package (can be transitive) referencing the dependency
+    # - reasonX should be the name of the invalid dependency
```

Comment:
I've refactored this constant so we can be more explicit about which integration is affected (previously only the parent dependency was taken into account)